### PR TITLE
TSDK-807 Added Support for Generating and Storing Peg-in Info

### PR DESCRIPTION
**Pull request by DiademShoukralla**
## Purpose

Goes with https://github.com/Topl/topl-btc-bridge/pull/15

Now that the PK is no longer hardcoded in the bridge, we need a way for users/investors to easily generate a new key. In addition, we need to store the key's information (so that it can be used at a later step, i.e, reclaiming).

## Approach

- Added 2 pages; "Get Public Key" and "Reclaim BTC". The former allows users to generate and store public key information (that will be used to reclaim funds). The latter allows users to reclaim pegged-in funds using the escrow address.
- Added the backend to support these use cases: Added SQLITE sb to persist data, added API end points, added bitcoin transaction&signature builders.

**Note to reviewer:** 
For Wallet Key Management, we opted for deriving a key via a descriptor instead of using a seed file (containing the mnemonic) because we were unable to associate the seed with a wallet (when creating via the RPC). We need this association since all wallet RPC calls require specifying a loaded wallet. it was non-trivial to connect a created wallet to a seed file; all the RPCs that allowed this (sethdseed, importprivkey, dumpwallet) only works on legacy wallets.

Basically, If I opted to do how the tbcli did it, then the wallets will not be loaded via the RPCs and we wont have some functionality (such as "get balances") on the demo wallet

## Testing

(locally) Ensured I was able to generate a new key, store its information, and then use it to reclaim the BTC.

After generating and storing:
Screenshot 2024-06-01 001327

After reclaiming:
Screenshot 2024-06-01 001233

I also tested attempting to reclaim the funds prior to 1000 blocks. It correctly errors out. 

## Tickets
* TSDK-807

### CODE REVIEW

**Comment by DiademShoukralla** [comment]
Merging without review since we will no longer be adding to this (or maintaining) this repo
